### PR TITLE
Feature/FPV-55 Changed fpId & sensorId field to mandatory

### PR DESCRIPTION
**Pull request by kendrick-807**
- Changed fingerprintId and sensorId fields to mandatory. They are required as parameters in the register route

### CODE REVIEW

File: routes/routes.js [hunk 1]
```diff
@@ -16,17 +16,27 @@ const JWT_KEY = process.env.JWT_SECRET;
 router.post('/register', async (req, res) => {
     const username = req.body.username;
     let password = req.body.password;
+    const fingerprintId = req.body.fingerprintId;
+    const sensorId = req.body.sensorId;
     if(!checkUserValidations(username, password)){
         logger.error("Username or password entered does not match pattern required!");
         return res.status(400).send({ error: "Username or password entered does not match pattern required!" });
     }
+    if(fingerprintId === undefined || fingerprintId === null || fingerprintId < 0 || isNaN(fingerprintId) ){
```
Comment (phamduylong) [inline routes/routes.js]: check with 162 as well :?

**Review by phamduylong — CHANGES_REQUESTED** [review]

File: routes/routes.js (hunk 1 above)
Comment (kendrick-807) [reply routes/routes.js]: oh yeah

File: routes/routes.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ router.post('/register', async (req, res) => {
         logger.error("Username or password entered does not match pattern required!");
         return res.status(400).send({ error: "Username or password entered does not match pattern required!" });
     }
-    if(fingerprintId === undefined || fingerprintId === null || fingerprintId < 0 || isNaN(fingerprintId) ){
+    if(fingerprintId === undefined || fingerprintId === null || fingerprintId < 0 || isNaN(fingerprintId) || fingerprintId > 162){
```
Comment (phamduylong) [inline routes/routes.js]: I'm thinking 161. We have 162 slots => 0-161

**Review by phamduylong — CHANGES_REQUESTED** [review]

**Comment by phamduylong** [comment]
@jaakkoiot @kikani-parth notice

File: routes/routes.js (hunk 2 above)
Comment (kendrick-807) [reply routes/routes.js]:
```js
fingerprintId: { type: Number, required: true, max: 162, default: -1 },
```
So this needs to be changed to 161 right @phamduylong

Comment (phamduylong) [reply routes/routes.js]: yep

**Comment by phamduylong** [comment]
image
Testing step failed somehow

**Comment by phamduylong** [comment]
Forgot to mention this on the ticket, but login should probably give a response body now. The fp id and sensor id needs to be sent to the device to check.

**Comment by kendrick-807** [comment]
> Forgot to mention this on the ticket, but login should probably give a response body now. The fp id and sensor id needs to be sent to the device to check.

Sure, I will change it as well

**Review by phamduylong — APPROVED** [review]

**Pull request merged**